app/models/provider.rb

```diff
@@ -310,14 +314,19 @@ def name_and_code
   end
 
   def accredited_bodies
-    accrediting_providers.map do |ap|
-      accrediting_provider_enrichment = accrediting_provider_enrichment(ap.provider_code)
-      {
-        provider_name: ap.provider_name,
-        provider_code: ap.provider_code,
-        description: accrediting_provider_enrichment&.Description || ''
-      }
-    end
+    accrediting_provider_enrichments&.filter_map do |accrediting_provider_enrichment|
+      provider_code = accrediting_provider_enrichment.UcasProviderCode
+
+      accredited_provider = recruitment_cycle.providers.find_by(provider_code:)
+
+      if accredited_provider.present?
+        {
```

Comment:
Should we return struct objects so it's a nicer API? Eg `AccreditedProvider = Struct.new(:provider_name, :provider_code, :description:, keyword_init: true)` or maybe even just a simple non activerecord model?

Reply:
I'm going to leave it as it is for now as its needed until we have replaced it as this is used on `Organisation eetails`

![image](https://github.com/DFE-Digital/publish-teacher-training/assets/470137/04d39be2-63cd-44f4-9258-cc66a59c6c5e)